src/components/ToggleSwitch/index.tsx

```diff
@@ -0,0 +1,206 @@
+    fontWeight: '400'
+  },
+  labelRight: {
+    marginLeft: 5,
+    color: '#FFFFFF',
+    fontSize: 12,
+    fontWeight: '400'
+  }
+});
+
+interface ToggleSwitchProps {
+  value?: boolean;
+  labelLeft?: string;
+  labelRight?: string;
+  circleActiveColor?: string;
+  circleInActiveColor?: string;
+  backgroundActive?: string;
+  backgroundInactive?: string;
+  styleLabelLeft?: StyleProp<TextStyle>;
+  styleLabelRight?: StyleProp<TextStyle>;
+  activeTextColor?: string;
+  inactiveTextColor?: string;
+  conatinerStyle?: TouchableOpacityProps['style'];
+  onValueChange?: (value: boolean) => void;
+}
+
+const ToggleSwitch: React.FC<ToggleSwitchProps> = ({
+  value = false,
+  onValueChange,
+  labelLeft,
+  labelRight,
+  circleActiveColor = '#00ADB5',
+  circleInActiveColor = '#2C67BC',
+  backgroundActive = '#F5F5F5',
+  backgroundInactive = '#F5F5F5',
+  styleLabelLeft,
+  styleLabelRight,
+  activeTextColor = '#00ADB5',
+  inactiveTextColor = '#2C67BC',
+  conatinerStyle
+}) => {
+  const positionButton = React.useMemo(() => new Animated.Value(0), []);
+
+  const startAnimToOff = () => {
+    Animated.timing(positionButton, {
+      toValue: 0,
+      duration: 500,
+      easing: Easing.ease,
+      useNativeDriver: false
+    }).start();
+  };
+
+  const startAnimToOn = () => {
+    Animated.timing(positionButton, {
+      toValue: 1,
+      duration: 500,
+      easing: Easing.ease,
+      useNativeDriver: false
+    }).start();
+  };
+
+  const positionInterPol = positionButton.interpolate({
+    inputRange: [0, 1],
+    outputRange: [0, dimen.normalizeDimen(22)]
+  });
+  const backgroundColorAnim = positionButton.interpolate({
+    inputRange: [0, 1],
+    outputRange: [backgroundInactive, backgroundActive]
+  });
+
+  const initialOpacityOn = positionButton.interpolate({inputRange: [0, 1], outputRange: [0, 1]});
+  const initialOpacityOff = positionButton.interpolate({inputRange: [0, 1], outputRange: [1, 0]});
+
+  const onPress = () => {
+    if (!value) {
+      startAnimToOff();
+    } else {
+      startAnimToOn();
+    }
+    if (onValueChange) onValueChange(!value);
+  };
+
+  if (value) {
+    startAnimToOn();
+  } else {
+    startAnimToOff();
+  }
+
+  return (
+    <TouchableOpacity
+      style={[styles.container, conatinerStyle]}
+      activeOpacity={0.9}
+      onPress={onPress}>
+      <Text style={[styles.labelLeft, styleLabelLeft]}>{labelLeft}</Text>
+      <Animated.View
+        style={[
+          styles.mainStyes,
+          {
+            backgroundColor: backgroundColorAnim
+          }
+        ]}>
+        <Animated.Text
+          style={[
+            styles.eahcStyles,
+            {
+              opacity: initialOpacityOn,
+              color: activeTextColor
+            }
+          ]}>
+          On
+        </Animated.Text>
+        <Animated.Text
+          style={[
+            styles.eahcStylesOf,
+            {
+              opacity: initialOpacityOff,
+              color: inactiveTextColor
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The hardcoded text 'On' and 'Off' may cause problems with localization. Consider passing these as props.
```diff
-           On
+           {labelOn}
        </Animated.Text>
        <Animated.Text
          style={[
            styles.eahcStylesOf,
            {
              opacity: initialOpacityOff,
              color: inactiveTextColor
            }
          ]}>
-           Off
+           {labelOff}
        </Animated.Text>
```
You would need to add `labelOn` and `labelOff` to your `ToggleSwitchProps` and destructure them in your component's props.



<!-- This is an auto-generated comment by OSS CodeRabbit -->